Keen.NetStandard.Test/QueryTests_Integration.cs

```diff
@@ -117,6 +121,1000 @@ class QueryTests_Integration : TestBase
+                queryParameters.Timeframe);
+
+            Assert.AreEqual(expectedResults.Count(), actualResults.Count());
+            var actualEnumerator = actualResults.GetEnumerator();
+            foreach (var expected in expectedResults)
+            {
+                actualEnumerator.MoveNext();
+                JToken actual = actualEnumerator.Current;
+                // Validate the result is correct
+                Assert.AreEqual(expected["user"]["email"].Value<string>(), actual["user"]["email"].Value<string>());
+                Assert.AreEqual(expected["user"]["id"].Value<string>(), actual["user"]["id"].Value<string>());
+                Assert.AreEqual(expected["user_agent"]["browser"].Value<string>(), actual["user_agent"]["browser"].Value<string>());
+            }
+        }
+
+        [Test]
+        public async Task Query_SimpleFunnel_Success()
+        {
+            var queryParameters = new FunnelParameters()
+            {
+                Steps = new FunnelStep[]
+                {
+                    new FunnelStep() {EventCollection = "signed up", ActorProperty = "visitor.guid", Timeframe = QueryRelativeTimeframe.ThisNDays(7)},
+                    new FunnelStep() {EventCollection = "completed profile", ActorProperty = "user.guid", Timeframe = QueryRelativeTimeframe.ThisNDays(7)},
+                    new FunnelStep() {EventCollection = "referred user", ActorProperty = "user.guid", Timeframe = QueryRelativeTimeframe.ThisNDays(7)},
+                }
+            };
+
+            string responseJson = @"{
+                ""result"": [
+                    3,
+                    1,
+                    0
+                ],
+                ""steps"": [
+                    {
+                        ""actor_property"": ""visitor.guid"",
+                        ""event_collection"": ""signed up"",
+                        ""timeframe"": ""this_7_days""
+                    },
+                    {
+                        ""actor_property"": ""user.guid"",
+                        ""event_collection"": ""completed profile"",
+                        ""timeframe"": ""this_7_days""
+                    },
+                    {
+                        ""actor_property"": ""user.guid"",
+                        ""event_collection"": ""referred user"",
+                        ""timeframe"": ""this_7_days""
+                    }
+                ]
+            }";
+
+            var expectedResponse = JObject.Parse(responseJson);
+
+            var client = CreateQueryTestKeenClient(queryParameters, expectedResponse);
+
+            var actualResults = await client.Queries.Funnel(
+                queryParameters.Steps);
+
+            var expectedResults = expectedResponse["result"];
+
+            Assert.AreEqual(expectedResults.Count(), actualResults.Result.Count());
+            var actualEnumerator = actualResults.Result.GetEnumerator();
+            foreach (var expected in expectedResults)
```

Comment:
This sort of iteration implies ordering...do we have reason to expect ordering will always be maintained? We saw that in a version to version change in Java where a particular collection was optimized and no longer maintained a certain ordering that some of the tests relied on, so it succeeded in Java 6 and 7 but broke when we tested on JDK 8 :/

Reply:
This is actually an ordered collection, so this should be fine.